plugins/woocommerce/includes/emails/class-wc-email.php

```diff
@@ -479,12 +486,37 @@ public function get_additional_content() {
 		return apply_filters( 'woocommerce_email_additional_content_' . $this->id, $this->format_string( $this->get_option_or_transient( 'additional_content' ) ), $this->object, $this );
 	}
 
+	/**
+	 * Ensure the personalization context is ready.
+	 *
+	 * This method ensures the Personalizer context is set correctly before emails are generated.
+	 * It skips context reconfiguration if it's already set.
+	 *
+	 * @return void
+	 */
+	protected function ensure_personalization_context_is_set(): void {
+		if ( $this->context_set ) {
+			// Context is already set, skip reconfiguration.
+			return;
+		}
```

Comment:
I think there might be an issue with this caching. Can we ensure we don't cache "empty" context?

The context data is set in trigger methods in classes extending WC_Email. So I have a concern that when somebody calls
`$email->getSubject()` before the trigger method is called, it may cause us to store an empty context, and we won't update it after the trigger is actually called.